### PR TITLE
Fix hot flow docs

### PR DESCRIPTION
**Pull request by hoc081098**
Should be MutableSharedFlow(replay=0) instead of MutableStateFlow(value=...)

### CODE REVIEW

**Comment by JakeWharton** [comment]
@kevincianfarini was the goal to use this type?

**Comment by kevincianfarini** [comment]
@JakeWharton This PR is correct. My bad. The original example I had written and tested in the issue which spawned this documentation is shared flow. 

Thanks @hoc081098!

**Comment by kevincianfarini** [comment]
(`replay` is only available for shared flow constructors, not state flow)

**Comment by JakeWharton** [comment]
Yeah I didn't know if the error was the it should have been `value=` or shared flow as the type.

File: README.md [hunk 1]
```diff
@@ -194,9 +194,9 @@ Emissions to hot flows that don't have active consumers are dropped. It's import
 (and therefore have an active collector) on a flow _before_ emissions to a flow are made. For example:
 
 ```kotlin
-val mutableStateFlow = MutableStateFlow<Int>(replay = 0)
-mutableStateFlow.emit(1)
-mutableStateFlow.test {
+val mutableSharedFlow = MutableSharedFlow<Int>(replay = 0)
```
Comment (kevincianfarini) [inline README.md:197]: Despite my original comments, can we change this to state flow? You can correct the param if you'd like. I imagine people use state flow more frequently than shared flow, and thus would like that to be the explicit example.

Comment (JakeWharton) [reply README.md:197]: Whoops too late. New PR?

Comment (kevincianfarini) [reply README.md:197]: @JakeWharton womp! Saw you merged. This isn't a big deal.

Comment (kevincianfarini) [reply README.md:197]: Yeah I can fix it up. One second. Pr incoming.

**Pull request merged**